Add blog post on `pulumi stack init --team` flag (#2588)
pulumi/pulumi-hugo: themes/default/content/blog/stack-init-teams-flag/index.md

---
title: "Announcing the Stack Init Teams Flag"

> date: "2023-03-28"

Reviewer: ```suggestion
date: "2023-03-31"
```

meta_desc: "We've added a new CLI flag to the stack init subcommand allowing users to grant team access to newly created stacks."
meta_image: "meta.png"

authors:
    - "robbie-mckinstry"
        
tags:
    - "cli"
    - "stack-init"
    - "cli-flags"
---

We recently released a quality-of-life feature for the Pulumi CLI to save users time and energy when creating stacks. Users can now initialize team permissions when creating stacks on the command line.

<!--more-->

For context, here's some background for any Pulumi newcomers. Pulumi projects

> consist of stacks—separate deployments of the same infrastructure. Stacks may

Author: I changed this to 
"stacks, which are…" 
which I think satisfies your suggestion. :)

also be configured with separate inputs. Organization admins can leverage RBAC
to grant project and stack access by sorting members into roles called "teams."

Stacks are often created via the CLI with the [`pulumi stack init` command](https://www.pulumi.com/docs/reference/cli/pulumi_stack_init/#options).
This command initializes a new stack. If you're using the
[Pulumi Service](https://www.pulumi.com/docs/intro/concepts/state/#pulumi-service-backend)
as your backend, you can view your newly created stack in the Service UI. If your
organization uses teams, you'll want to give your teammates access to the stack
you created.

With the release of Pulumi v3.59.0, developers can assign team access during stack creation. To do this,
pass in the `--team` flag followed by the name of the team. For instance:

> `pulumi stack init --team Red`. This saves time by eliminating context switching,

Reviewer: GitHub doesn't make it easy to make this sort of suggestion, but I'd love to see this called out on a line by itself, since it's really the whole story of the blog post -- something like:

--

With the release of Pulumi v3.59.0, developers can assign team access during stack creation. To do this,
pass in the `--team` flag followed by the name of the team. For instance:

```
pulumi stack init --team Red
```

This saves time by eliminating context switching...

--

I'd suggest the same for the example below this one that shows multiple flags in the same call.

> allowing users to stay on task. Previously, the only way to assign team access to

Author: Ah, yeah, I can adjust this so it doesn't falsely claim the UI was the only way to assign permissions, but I don't want to get into the weeds listing all of the ways this can be done.

> a stack was through the Service UI. This meant a necessary trip to a browser

Author: I retouched this one as well.

> before your teammates could peek at your work. This enhancement streamlines the

Reviewer 2: ```suggestion
This enhancement streamlines the
```

> DX of stack creation, helping you spin up new infrastructure.

Reviewer 2: ```suggestion
process of managing stack permissions without navigating elsewhere.```

> Use the flag multiple times to assign access to multiple teams, as in

Reviewer 2: ```suggestion
Use the flag multiple times to assign access to multiple Pulumi Teams, as in
```

`pulumi stack init --teams Red --teams Blue`. Currently, the feature always grants
`read` and `write` access, the most commonly assigned permissions.

> We implemented this feature at the request of our daily users.

Reviewer 2: ```suggestion
We implemented this feature as a direct result of our customers feedback.
```

> Context switching can be expensive when you're creating enough stacks.

Reviewer 2: ```suggestion
```

At Pulumi, we enable our users to squeak out as much productivity from our tooling
as possible. We're always happy to make ergonomic improvements. To those of you

> who want to save every keystroke, check out our Automation API for programmatic

Reviewer: Suggest a link for this -- although TBH, it feels a bit bolted on and not really related to the content of this post, so IMO removing this sentence altogether would probably be okay, too:

```suggestion
who want to save every keystroke, check out our [Automation API](/docs/guides/automation-api/) for programmatic
```

Author: Who doesn't love a good cross-sell? ;D
Nah, I'm only joking. :-) You're right, I think it's better without the sentence entirely. Good call, Christian.

---
title: "Announcing the Stack Init Teams Flag"
date: "2023-03-28"
meta_desc: "We've added a new CLI flag to the stack init subcommand allowing users to grant team access to newly created stacks."
meta_image: "meta.png"

authors:
    - "robbie-mckinstry"
        
tags:
    - "cli"
    - "stack-init"
    - "cli-flags"
---

We recently released a quality-of-life feature for the Pulumi CLI to save users time and energy when creating stacks. Users can now initialize team permissions when creating stacks on the command line.

<!--more-->

For context, here's some background for any Pulumi newcomers. Pulumi projects
consist of stacks—separate deployments of the same infrastructure. Stacks may
also be configured with separate inputs. Organization admins can leverage RBAC

> to  grant project and stack access by sorting members into roles called "teams."

Reviewer 3: ```suggestion
to grant project and stack access by sorting members into roles called "teams."
```

Reviewer 3: is this really how we tell users to think about this? that teams are roles? 🤔

Author: @susanev that's how _I_ think about this! Maybe marketing and product feel differently!
But when I think of Pulumi Teams, I think of RBAC. It looks like that's how we pitch it, too:
https://www.pulumi.com/docs/intro/pulumi-service/teams/
> # Teams and RBAC
> The Pulumi Service offers role-based access control (RBAC) using teams. Teams allow organization admins to assign a set of stack permissions to a group of users.

> Most often, stacks are created via the CLI with the `pulumi stack init` command.
> [This command](https://www.pulumi.com/docs/reference/cli/pulumi_stack_init/#options)
> initializes a new stack. If you're using the

Reviewer 3: ```suggestion
Stacks are often created via the CLI with the [`pulumi stack init` command](https://www.pulumi.com/docs/reference/cli/pulumi_stack_init/#options).
This command initializes a new stack. If you're using the
```
